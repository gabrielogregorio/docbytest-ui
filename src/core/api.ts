/* eslint-disable no-unused-vars */
/* eslint-disable @typescript-eslint/no-unused-vars */
/* eslint-disable import/no-unresolved */
import { apiMethodType } from './interfaces/api';

type apiType = {
  url: string;
  method: apiMethodType;
  body: any;
};

export const Api = async ({ url, method }: apiType) => {
  return fetch(url, {
    method,
    // body,
    // mode?: "navigate" | "same-origin" | "no-cors" | "cors",
    headers: undefined, // {'Content-type': 'application/json; charset=UTF-8',},
  });
};
