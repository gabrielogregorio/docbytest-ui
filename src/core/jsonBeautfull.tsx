import { generateIds } from './shared/generateIds';

const colors = {
  special: 'text-purple-300',
  string: 'text-orange-300',
  keyString: 'text-blue-300',
  default: 'text-gray-300',
};

const DivideJson = () => {
  return <span className="text-gray-300">:</span>;
};

const KeyString = ({ text }: { text: string }) => {
  return <span className={colors.keyString}>{text}</span>;
};

const CommaTreatment = ({ value, colorsLocal }: { value: string; colorsLocal: string }) => {
  const lastCharacter = value.slice(value.length - 1, value.length - 1 + 1);
  const textWithoutLastCharacter = value.slice(0, value.length - 1);

  if (lastCharacter === ',') {
    return (
      <>
        <span className={`${colorsLocal}`}>{textWithoutLastCharacter}</span>
        <span className={colors.default}>,</span>
      </>
    );
  }

  return <span className={`${colorsLocal}`}>{value}</span>;
};

export const JsonColorizer = ({ text }: { text: string }) => {
  const linesText = text.split('\n');

  const renderLines = () => {
    return linesText.map((lineText) => {
      const line = lineText?.toString() || '';
      const [key, value] = line.split(':');

      const lineContainsOnlySpecialCharacters = ['{', '[', '}', ']', '},', '],'].includes(line.trim());
      if (lineContainsOnlySpecialCharacters) {
        return (
          <div key={generateIds()}>
            <CommaTreatment value={line} colorsLocal={colors.default} />
          </div>
        );
      }

      if (!key || !value) {
        return (
          <div key={generateIds()}>
            <span className={colors.default}>{line}</span>
          </div>
        );
      }

      const valueContainOnlySpecialTypes = /(^\d*|false|true|null)[,|]$/.test(value.trim());
      if (valueContainOnlySpecialTypes) {
        return (
          <div key={generateIds()}>
            <KeyString text={key} />
            <DivideJson />
            <CommaTreatment value={value} colorsLocal={colors.special} />
          </div>
        );
      }

      const valueContainsOnlyWithSpecialCharacters = ['[', '{'].includes(value.trim());
      if (valueContainsOnlyWithSpecialCharacters) {
        return (
          <div key={generateIds()}>
            <KeyString text={key} />
            <DivideJson />
            <CommaTreatment value={value} colorsLocal={colors.default} />
          </div>
        );
      }

      return (
        <div key={generateIds()}>
          <KeyString text={key} />
          <DivideJson />
          <CommaTreatment value={value} colorsLocal={colors.string} />
        </div>
      );
    });
  };

  return <>{renderLines()}</>;
};
