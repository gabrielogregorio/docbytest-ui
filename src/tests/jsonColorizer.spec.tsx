import { render } from '@testing-library/react';
import { JsonColorizer } from '../core/jsonBeautfull';

test('Should colorize json', () => {
  const { container } = render(
    <JsonColorizer
      text={JSON.stringify(
        {
          title: 'title content',
          user: '',
          item: null,
          count: 1,
          type: {
            false: false,
            true: true,
          },
          imgs: [
            {
              id: '1',
              img: 'img/pontinho.png',
            },
          ],
        },
        null,
        4,
      )}
    />,
  );

  expect(container).toMatchSnapshot();
});
