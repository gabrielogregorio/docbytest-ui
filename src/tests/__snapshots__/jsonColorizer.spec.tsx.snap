// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should colorize json 1`] = `
<div>
  <div>
    <span
      class="text-gray-300"
    >
      {
    </span>
  </div>
  <div>
    <span
      class="text-blue-300"
    >
          "title"
    </span>
    <span
      class="text-gray-300"
    >
      :
    </span>
    <span
      class="text-orange-300"
    >
       "title content"
    </span>
    <span
      class="text-gray-300"
    >
      ,
    </span>
  </div>
  <div>
    <span
      class="text-blue-300"
    >
          "user"
    </span>
    <span
      class="text-gray-300"
    >
      :
    </span>
    <span
      class="text-orange-300"
    >
       ""
    </span>
    <span
      class="text-gray-300"
    >
      ,
    </span>
  </div>
  <div>
    <span
      class="text-blue-300"
    >
          "item"
    </span>
    <span
      class="text-gray-300"
    >
      :
    </span>
    <span
      class="text-purple-300"
    >
       null
    </span>
    <span
      class="text-gray-300"
    >
      ,
    </span>
  </div>
  <div>
    <span
      class="text-blue-300"
    >
          "count"
    </span>
    <span
      class="text-gray-300"
    >
      :
    </span>
    <span
      class="text-purple-300"
    >
       1
    </span>
    <span
      class="text-gray-300"
    >
      ,
    </span>
  </div>
  <div>
    <span
      class="text-blue-300"
    >
          "type"
    </span>
    <span
      class="text-gray-300"
    >
      :
    </span>
    <span
      class="text-gray-300"
    >
       {
    </span>
  </div>
  <div>
    <span
      class="text-blue-300"
    >
              "false"
    </span>
    <span
      class="text-gray-300"
    >
      :
    </span>
    <span
      class="text-purple-300"
    >
       false
    </span>
    <span
      class="text-gray-300"
    >
      ,
    </span>
  </div>
  <div>
    <span
      class="text-blue-300"
    >
              "true"
    </span>
    <span
      class="text-gray-300"
    >
      :
    </span>
    <span
      class="text-orange-300"
    >
       true
    </span>
  </div>
  <div>
    <span
      class="text-gray-300"
    >
          }
    </span>
    <span
      class="text-gray-300"
    >
      ,
    </span>
  </div>
  <div>
    <span
      class="text-blue-300"
    >
          "imgs"
    </span>
    <span
      class="text-gray-300"
    >
      :
    </span>
    <span
      class="text-gray-300"
    >
       [
    </span>
  </div>
  <div>
    <span
      class="text-gray-300"
    >
              {
    </span>
  </div>
  <div>
    <span
      class="text-blue-300"
    >
                  "id"
    </span>
    <span
      class="text-gray-300"
    >
      :
    </span>
    <span
      class="text-orange-300"
    >
       "1"
    </span>
    <span
      class="text-gray-300"
    >
      ,
    </span>
  </div>
  <div>
    <span
      class="text-blue-300"
    >
                  "img"
    </span>
    <span
      class="text-gray-300"
    >
      :
    </span>
    <span
      class="text-orange-300"
    >
       "img/pontinho.png"
    </span>
  </div>
  <div>
    <span
      class="text-gray-300"
    >
              }
    </span>
  </div>
  <div>
    <span
      class="text-gray-300"
    >
          ]
    </span>
  </div>
  <div>
    <span
      class="text-gray-300"
    >
      }
    </span>
  </div>
</div>
`;
